### PR TITLE
Wrap errors from deferred Close() calls

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: db/db_test.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+func TestCloseAndJoinOnError(t *testing.T) {
+	var err error
+	defer func() {
+		db.CloseAndWrapOnError(errorer{}, &err)
+		assert.EqualError(t, errors.Unwrap(err), "closer error")
+	}()
+
+	db.CloseAndWrapOnError(errorer{}, &err)
+	assert.EqualError(t, err, "closer error")
```
[the remark this answers is not in the document]
Comment (omerfirmak) [reply db/db_test.go:25]: It just tests 2 branches in CloseAndWrapOnError, with and without an existing error

File: db/db.go [hunk 2]
```diff
@@ -73,4 +74,17 @@ type Transaction interface {
...
+}
+
+func CloseAndWrapOnError(closer io.Closer, existingErr *error) {
+	if closeErr := closer.Close(); closeErr != nil {
+		if *existingErr != nil {
+			*existingErr = fmt.Errorf(`failed to close because "%v"" with existing err "%w"`, closeErr, *existingErr)
+		} else {
+			*existingErr = closeErr
+		}
+	}
```
[the remark this answers is not in the document]
Comment (omerfirmak) [reply db/db.go:89]: Much better, thanks!

File: db/db_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (omerfirmak) [reply db/db_test.go]: and makes sure it works as expected in `defer`

[the remark this answers is not in the document]
Comment (IronGauntlets) [reply db/db_test.go]: I see I didn't notice the inputs were different. In that case please name them through `t.Run` as separate cases.

**Review by IronGauntlets — APPROVED** [review]

**Pull request merged**